--- a/plasma/netbook/shell/plasma-netbook.desktop
+++ b/plasma/netbook/shell/plasma-netbook.desktop
@@ -109,4 +109,4 @@
 X-KDE-StartupNotify=false
 OnlyShowIn=KDE;
 X-KDE-autostart-phase=0
-
+Hidden=true
--- a/plasma/netbook/shell/CMakeLists.txt
+++ b/plasma/netbook/shell/CMakeLists.txt
@@ -22,5 +22,5 @@
 install(TARGETS kdeinit_plasma-netbook DESTINATION ${LIB_INSTALL_DIR})
 install(TARGETS plasma-netbook ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES plasma-default-layoutrc DESTINATION ${DATA_INSTALL_DIR}/plasma-netbook/)
-#install(FILES plasma-netbook.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
+install(FILES plasma-netbook.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
 
