--- workspace/kcontrol/dateandtime/clock.desktop	(revision 725651)
+++ workspace/kcontrol/dateandtime/clock.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kde-clock.desktop
+Exec=kcmshell4 kde-clock.desktop
 Icon=preferences-system-time
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/background/background.desktop	(revision 725651)
+++ workspace/kcontrol/background/background.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell background
+Exec=kcmshell4 background
 Icon=preferences-desktop-wallpaper
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/kfontinst/kcmfontinst/fontinst.desktop	(revision 725651)
+++ workspace/kcontrol/kfontinst/kcmfontinst/fontinst.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell fontinst
+Exec=kcmshell4 fontinst
 Icon=preferences-desktop-font-installer
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/screensaver/screensaver.desktop	(revision 725651)
+++ workspace/kcontrol/screensaver/screensaver.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell screensaver
+Exec=kcmshell4 screensaver
 Icon=preferences-desktop-wallpaper
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/smartcard/smartcard.desktop	(revision 725651)
+++ workspace/kcontrol/smartcard/smartcard.desktop	(revision 725652)
@@ -3,7 +3,7 @@
 Icon=preferences-desktop-user-smartcard
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
-Exec=kcmshell smartcard
+Exec=kcmshell4 smartcard
 DocPath=kcontrol/smartcard.html
 
 X-KDE-Library=kcm_smartcard
--- workspace/kcontrol/accessibility/accessibility.desktop	(revision 725651)
+++ workspace/kcontrol/accessibility/accessibility.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell accessibility
+Exec=kcmshell4 accessibility
 Icon=preferences-desktop-accessibility
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/taskbar/kcmtaskbar.desktop	(revision 725651)
+++ workspace/kcontrol/taskbar/kcmtaskbar.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kcmtaskbar
+Exec=kcmshell4 kcmtaskbar
 Icon=preferences-desktop-panel-taskbar
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/fonts/fonts.desktop	(revision 725651)
+++ workspace/kcontrol/fonts/fonts.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell fonts
+Exec=kcmshell4 fonts
 Icon=preferences-desktop-font
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/energy/energy.desktop	(revision 725651)
+++ workspace/kcontrol/energy/energy.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell energy
+Exec=kcmshell4 energy
 Icon=preferences-system-power-management
 DocPath=kcontrol/energy/index.html
 Type=Service
--- workspace/kcontrol/infocenter/usbview/kcmusb.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/usbview/kcmusb.desktop	(revision 725652)
@@ -4,7 +4,7 @@
 Type=Service
 ServiceTypes=KCModule
 DocPath=kinfocenter/usb/index.html
-Exec=kcmshell kcmusb
+Exec=kcmshell4 kcmusb
 
 X-KDE-Library=kcm_usb
 X-KDE-ParentApp=kinfocenter
--- workspace/kcontrol/infocenter/info/processor.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/processor.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell processor
+Exec=kcmshell4 processor
 Icon=kcmprocessor
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/devices.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/devices.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell devices
+Exec=kcmshell4 devices
 Icon=kcmdevices
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/sound.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/sound.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell sound
+Exec=kcmshell4 sound
 Icon=kcmsound
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/partitions.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/partitions.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell partitions
+Exec=kcmshell4 partitions
 Icon=kcmpartitions
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/xserver.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/xserver.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell xserver
+Exec=kcmshell4 xserver
 Icon=kcmx
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/ioports.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/ioports.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell ioports
+Exec=kcmshell4 ioports
 Icon=kcmmemory
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/dma.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/dma.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell dma
+Exec=kcmshell4 dma
 Icon=kcmmemory
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/memory.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/memory.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell memory
+Exec=kcmshell4 memory
 Icon=kcmmemory
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/pci.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/pci.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell pci
+Exec=kcmshell4 pci
 Icon=kcmpci
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/interrupts.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/interrupts.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell interrupts
+Exec=kcmshell4 interrupts
 Icon=kcmmemory
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/opengl.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/opengl.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell opengl
+Exec=kcmshell4 opengl
 Icon=kcmopengl
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/info/scsi.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/info/scsi.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell scsi
+Exec=kcmshell4 scsi
 Icon=kcmscsi
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/infocenter/view1394/kcmview1394.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/view1394/kcmview1394.desktop	(revision 725652)
@@ -4,7 +4,7 @@
 Type=Service
 ServiceTypes=KCModule
 DocPath=kinfocenter/view1394/index.html
-Exec=kcmshell kcmview1394
+Exec=kcmshell4 kcmview1394
 
 X-KDE-Library=kcm_view1394
 X-KDE-ParentApp=kinfocenter
--- workspace/kcontrol/infocenter/nics/nic.desktop	(revision 725651)
+++ workspace/kcontrol/infocenter/nics/nic.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell nic
+Exec=kcmshell4 nic
 Type=Service
 ServiceTypes=KCModule
 DocPath=kinfocenter/nics/index.html
--- workspace/kcontrol/hardware/joystick/joystick.desktop	(revision 725651)
+++ workspace/kcontrol/hardware/joystick/joystick.desktop	(revision 725652)
@@ -60,7 +60,7 @@
 Comment[x-test]=xxjoystick - a kcontrol module to test joysticksxx
 Comment[zh_CN]=joystick - 测试游戏杆的 kcontrol 模块
 Comment[zh_TW]=搖桿 - 用來測試搖桿的 kcontrol 模組
-Exec=kcmshell joystick
+Exec=kcmshell4 joystick
 Keywords=joystick,gamepad
 Keywords[ar]=وسادة اللعبة,عصا الألعاب
 Keywords[be]=Джойстык,joystick,gamepad
--- workspace/kcontrol/hardware/display/display.desktop	(revision 725651)
+++ workspace/kcontrol/hardware/display/display.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell display
+Exec=kcmshell4 display
 Icon=preferences-desktop-display
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/input/mouse.desktop	(revision 725651)
+++ workspace/kcontrol/input/mouse.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell mouse
+Exec=kcmshell4 mouse
 Icon=preferences-desktop-mouse
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/kcontrol/kxkb/kcmmisc/keyboard.desktop	(revision 725651)
+++ workspace/kcontrol/kxkb/kcmmisc/keyboard.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell keyboard
+Exec=kcmshell4 keyboard
 Icon=preferences-desktop-keyboard
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/kcontrol/kxkb/keyboard_layout.desktop	(revision 725651)
+++ workspace/kcontrol/kxkb/keyboard_layout.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell keyboard_layout
+Exec=kcmshell4 keyboard_layout
 Icon=preferences-desktop-keyboard-layout
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/kxkb/kxkbcore.cpp	(revision 725651)
+++ workspace/kcontrol/kxkb/kxkbcore.cpp	(revision 725652)
@@ -346,7 +346,7 @@
     {
         QStringList lst;
         lst<< "keyboard_layout";
-	QProcess::startDetached("kcmshell",lst);
+	QProcess::startDetached("kcmshell4",lst);
     }
     else if (id == KxkbWidget::HELP_MENU_ID)
     {
--- workspace/kcontrol/style/style.desktop	(revision 725651)
+++ workspace/kcontrol/style/style.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell style
+Exec=kcmshell4 style
 Icon=preferences-desktop-theme-style
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/kcontrol/colors/colors.desktop	(revision 725651)
+++ workspace/kcontrol/colors/colors.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell colors
+Exec=kcmshell4 colors
 Icon=preferences-desktop-color
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/access/kcmaccess.desktop	(revision 725651)
+++ workspace/kcontrol/access/kcmaccess.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kcmaccess
+Exec=kcmshell4 kcmaccess
 Icon=preferences-desktop-accessibility
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/kcontrol/xinerama/xinerama.desktop	(revision 725651)
+++ workspace/kcontrol/xinerama/xinerama.desktop	(revision 725652)
@@ -2,7 +2,7 @@
 Icon=preferences-desktop-display-multiple
 Type=Service
 ServiceTypes=KCModule
-Exec=kcmshell xinerama
+Exec=kcmshell4 xinerama
 DocPath=kcontrol/multiplemonitors.html
 Hidden[$e]=$(if test_kcm_xinerama; then echo "false"; else echo "true"; fi)
 
--- workspace/kcontrol/randr/randr.desktop	(revision 725651)
+++ workspace/kcontrol/randr/randr.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell randr
+Exec=kcmshell4 randr
 Icon=preferences-desktop-display-randr
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/smserver/kcmsmserver.desktop	(revision 725651)
+++ workspace/kcontrol/smserver/kcmsmserver.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kcmsmserver
+Exec=kcmshell4 kcmsmserver
 Icon=preferences-system-session
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/kthememanager/kthememanager.desktop	(revision 725651)
+++ workspace/kcontrol/kthememanager/kthememanager.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kthememanager
+Exec=kcmshell4 kthememanager
 Icon=preferences-desktop-theme-manager
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/kthememanager/kthemedlg.ui.h	(revision 725651)
+++ workspace/kcontrol/kthememanager/kthemedlg.ui.h	(revision 725652)
@@ -19,32 +19,32 @@
 
 void KThemeDlg::startBackground()
 {
-    KRun::runCommand("kcmshell background", this);
+    KRun::runCommand("kcmshell4 background", this);
 }
 
 void KThemeDlg::startColors()
 {
-    KRun::runCommand("kcmshell colors", this);
+    KRun::runCommand("kcmshell4 colors", this);
 }
 
 void KThemeDlg::startStyle()
 {
-    KRun::runCommand("kcmshell style", this);
+    KRun::runCommand("kcmshell4 style", this);
 }
 
 void KThemeDlg::startIcons()
 {
-    KRun::runCommand("kcmshell icons", this);
+    KRun::runCommand("kcmshell4 icons", this);
 }
 
 void KThemeDlg::startFonts()
 {
-   KRun::runCommand("kcmshell fonts", this);
+   KRun::runCommand("kcmshell4 fonts", this);
 }
 
 void KThemeDlg::startSaver()
 {
-    KRun::runCommand("kcmshell screensaver", this);
+    KRun::runCommand("kcmshell4 screensaver", this);
 }
 
 #endif // KTHEMEDLG_UI_H
--- workspace/kcontrol/launch/kcmlaunch.desktop	(revision 725651)
+++ workspace/kcontrol/launch/kcmlaunch.desktop	(revision 725652)
@@ -4,7 +4,7 @@
 Type=Service
 ServiceTypes=KCModule
 DocPath=kcontrol/kcmlaunch/index.html
-Exec=kcmshell kcmlaunch
+Exec=kcmshell4 kcmlaunch
 
 X-KDE-Library=kcm_launch
 X-KDE-ParentApp=kcontrol
--- workspace/kcontrol/keys/keys.desktop	(revision 725651)
+++ workspace/kcontrol/keys/keys.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell keys
+Exec=kcmshell4 keys
 Icon=preferences-desktop-keyboard-shortcut
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/kcontrol/kdm/kdm.desktop	(revision 725651)
+++ workspace/kcontrol/kdm/kdm.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kdm
+Exec=kcmshell4 kdm
 Icon=preferences-desktop-login
 Type=Service
 ServiceTypes=KCModule
--- workspace/kcontrol/bell/bell.desktop	(revision 725651)
+++ workspace/kcontrol/bell/bell.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell bell
+Exec=kcmshell4 bell
 Icon=preferences-desktop-notification-bell
 Type=Service
 ServiceTypes=KCModule,KCModuleInit
--- workspace/solid/kcm/kcm_solid.desktop	(revision 725651)
+++ workspace/solid/kcm/kcm_solid.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kcm_solid
+Exec=kcmshell4 kcm_solid
 Icon=preferences-system-solid
 Type=Service
 ServiceTypes=KCModule
--- workspace/khotkeys/kcontrol/khotkeys.desktop	(revision 725651)
+++ workspace/khotkeys/kcontrol/khotkeys.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell khotkeys
+Exec=kcmshell4 khotkeys
 Icon=preferences-desktop-keyboard-khotkeys
 Type=Service
 ServiceTypes=KCModule
--- workspace/ksplash/kcm/ksplashthememgr.desktop	(revision 725651)
+++ workspace/ksplash/kcm/ksplashthememgr.desktop	(revision 725652)
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Icon=preferences-system-splash
-Exec=kcmshell ksplashthememgr
+Exec=kcmshell4 ksplashthememgr
 DocPath=ksplashml/themes.html
 
 Type=Service
--- workspace/kwin/kcmkwin/kwindecoration/kwindecoration.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwindecoration/kwindecoration.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kwindecoration
+Exec=kcmshell4 kwindecoration
 Icon=preferences-system-windows-decoration
 Type=Service
 ServiceTypes=KCModule
--- workspace/kwin/kcmkwin/kwinrules/kwinrules.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinrules/kwinrules.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kwinrules
+Exec=kcmshell4 kwinrules
 Icon=preferences-system-windows-specific
 Type=Service
 ServiceTypes=KCModule
--- workspace/kwin/kcmkwin/kwincompositing/kwincompositing.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwincompositing/kwincompositing.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kwincompositing
+Exec=kcmshell4 kwincompositing
 Icon=preferences-system-windows-effects
 Type=Service
 ServiceTypes=KCModule
--- workspace/kwin/kcmkwin/kwinoptions/kwinfocus.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinoptions/kwinfocus.desktop	(revision 725652)
@@ -3,7 +3,7 @@
 Icon=kcmkwm
 Type=Service
 ServiceTypes=KCModule
-Exec=kcmshell kwinfocus
+Exec=kcmshell4 kwinfocus
 DocPath=kcontrol/windowmanagement/index.html#action-focus
 
 X-KDE-Library=kcm_kwinoptions
--- workspace/kwin/kcmkwin/kwinoptions/kwinoptions.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinoptions/kwinoptions.desktop	(revision 725652)
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Exec=kcmshell kwinoptions
+Exec=kcmshell4 kwinoptions
 Icon=preferences-system-windows-behaviour
 Type=Service
 ServiceTypes=KCModule
--- workspace/kwin/kcmkwin/kwinoptions/kwinmoving.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinoptions/kwinmoving.desktop	(revision 725652)
@@ -3,7 +3,7 @@
 Icon=kcmkwm
 Type=Service
 ServiceTypes=KCModule
-Exec=kcmshell kwinmoving
+Exec=kcmshell4 kwinmoving
 DocPath=kcontrol/windowmanagement/index.html#action-moving
 
 X-KDE-Library=kcm_kwinoptions
--- workspace/kwin/kcmkwin/kwinoptions/kwinactions.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinoptions/kwinactions.desktop	(revision 725652)
@@ -3,7 +3,7 @@
 Icon=kcmkwm
 Type=Service
 ServiceTypes=KCModule
-Exec=kcmshell kwinactions
+Exec=kcmshell4 kwinactions
 DocPath=kcontrol/windowmanagement/index.html#action-actions
 
 X-KDE-Library=kcm_kwinoptions
--- workspace/kwin/kcmkwin/kwinoptions/kwinadvanced.desktop	(revision 725651)
+++ workspace/kwin/kcmkwin/kwinoptions/kwinadvanced.desktop	(revision 725652)
@@ -3,7 +3,7 @@
 Icon=kcmkwm
 Type=Service
 ServiceTypes=KCModule
-Exec=kcmshell kwinadvanced
+Exec=kcmshell4 kwinadvanced
 DocPath=kcontrol/windowmanagement/index.html#action-advanced
 
 X-KDE-Library=kcm_kwinoptions
--- workspace/kwin/workspace.cpp	(revision 725651)
+++ workspace/kwin/workspace.cpp	(revision 725652)
@@ -1140,7 +1140,7 @@
 
 void Workspace::configureWM()
     {
-	KToolInvocation::kdeinitExec( "kcmshell", configModules(false) );
+	KToolInvocation::kdeinitExec( "kcmshell4", configModules(false) );
     }
 
 /*!
--- workspace/config-workspace.h.cmake	(revision 726654)
+++ workspace/config-workspace.h.cmake	(revision 726655)
@@ -91,13 +91,13 @@
 #cmakedefine HAVE_VSNPRINTF 1
 
 /* KDE's binaries directory */
-#define KDE_BINDIR "${CMAKE_INSTALL_PREFIX}/bin"
+#define KDE_BINDIR "${BIN_INSTALL_DIR}"
 
 /* KDE's configuration directory */
-#define KDE_CONFDIR "${CMAKE_INSTALL_PREFIX}/share/config"
+#define KDE_CONFDIR "${CONFIG_INSTALL_DIR}"
 
 /* KDE's static data directory */
-#define KDE_DATADIR "${CMAKE_INSTALL_PREFIX}/share/apps"
+#define KDE_DATADIR "${DATA_INSTALL_DIR}"
 
 /* Define where your java executable is */
 #undef PATH_JAVA
