Index: workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.h
===================================================================
--- workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.h	(revisione 922899)
+++ workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.h	(copia locale)
@@ -52,11 +52,11 @@
         bool isConnected(QString id);
         bool isActivated(QString id);
         int rotation(QString id);
-        qreal rate(QString id);
+        double rate(QString id);
         bool reflectX(QString id);
         bool reflectY(QString id);
         int numAvailableRates(QString id);
-        qreal availableRate(QString id, int index);
+        double availableRate(QString id, int index);
         
     Q_SIGNALS:
         void outputConnected(QString id);
Index: workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.cpp
===================================================================
--- workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.cpp	(revisione 922899)
+++ workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.cpp	(copia locale)
@@ -109,10 +109,10 @@
     return false;
 }
 
-bool DBusAPIConfigurations::changeRate(QString output, qreal rate) {
+bool DBusAPIConfigurations::changeRate(QString output, double rate) {
     Output * o = Outputs::self()->output(output);
     if (o) {
-        return Configurations::self()->changeRate(o, rate);
+        return Configurations::self()->changeRate(o, static_cast<float>(rate));
     }
     return false;
 }
Index: workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.h
===================================================================
--- workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.h	(revisione 922899)
+++ workspace/libs/kephal/kded_kephal/dbus/dbusapi_configurations.h	(copia locale)
@@ -51,7 +51,7 @@
         bool move(QString output, QPoint position);
         bool resize(QString output, QSize size);
         bool rotate(QString output, int rotation);
-        bool changeRate(QString output, qreal rate);
+        bool changeRate(QString output, double rate);
         bool reflectX(QString output, bool reflect);
         bool reflectY(QString output, bool reflect);
         int screen(QString output);
Index: workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.cpp
===================================================================
--- workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.cpp	(revisione 922899)
+++ workspace/libs/kephal/kded_kephal/dbus/dbusapi_outputs.cpp	(copia locale)
@@ -127,7 +127,7 @@
     return 0;
 }
 
-qreal DBusAPIOutputs::availableRate(QString id, int i)
+double DBusAPIOutputs::availableRate(QString id, int i)
 {
     if (! m_rates.contains(id)) {
         numAvailableRates(id);
@@ -146,7 +146,7 @@
     return RotateNormal;
 }
 
-qreal DBusAPIOutputs::rate(QString id) {
+double DBusAPIOutputs::rate(QString id) {
     Output * output = Outputs::self()->output(id);
     if (output && output->isActivated()) {
         return output->rate();
