Index: workspace/config-workspace.h.cmake
===================================================================
--- workspace/config-workspace.h.cmake	(revision 725575)
+++ workspace/config-workspace.h.cmake	(working copy)
@@ -94,7 +94,7 @@
 #define KDE_BINDIR "${CMAKE_INSTALL_PREFIX}/bin"
 
 /* KDE's configuration directory */
-#define KDE_CONFDIR "${CMAKE_INSTALL_PREFIX}/share/config"
+#define KDE_CONFDIR "${SYSCONF_INSTAL_DIR}/kde4"
 
 /* KDE's static data directory */
 #define KDE_DATADIR "${CMAKE_INSTALL_PREFIX}/share/apps"
