Index: kdebase-workspace/kdm/kfrontend/genkdmconf.c
===================================================================
--- kdebase-workspace.orig/kdm/kfrontend/genkdmconf.c	2010-12-18 20:44:22.682071103 +0100
+++ kdebase-workspace/kdm/kfrontend/genkdmconf.c	2010-12-18 20:44:27.662318969 +0100
@@ -3141,9 +3141,11 @@
                 memcmp(oldkde + olen - sizeof(SHR_CONF) + 1,
                        SHR_CONF, sizeof(SHR_CONF)))
             {
+                /*
                 fprintf(stderr,
                         "Warning: --old-kde does not end with " SHR_CONF ". "
                         "Might wreak havoc.\n");
+                */
                 oldkdepfx = oldkde;
             } else
                 ASPrintf((char **)&oldkdepfx,
