diff -Naur kdebase-workspace.orig/kdm/config.def kdebase-workspace.patched/kdm/config.def
--- kdebase-workspace.orig/kdm/config.def	2007-09-15 13:31:20.000000000 +0200
+++ kdebase-workspace.patched/kdm/config.def	2007-09-15 13:48:04.000000000 +0200
@@ -2657,7 +2657,7 @@
 Type: string
 Default: ""
 User: greeter
-Instance: */KDMDATA "/themes/circles"
+Instance: */ "@@@ToBeReplacedByDesktopBase@@@"
 Comment: &
 Description:
  The theme to use for the greeter. Can point to either a directory or an XML
