diff -Naur kdebase-workspace.orig/kdm/config.def kdebase-workspace.patched/kdm/config.def
--- kdebase-workspace.orig/kdm/config.def	2007-09-15 13:31:20.000000000 +0200
+++ kdebase-workspace.patched/kdm/config.def	2007-09-16 15:16:15.000000000 +0200
@@ -90,6 +90,12 @@
 <kdmrc>
  &kdm; master configuration file
  _
+ Please note: Settings in this file are sometimes ignored (overridden).
+ The default KDM startup script /etc/init.d/kdm looks in /etc/default/kdm.d
+ for theme-related settings which, if found, take precedence. The possibly 
+ overridden settings are: UseBackground, BackgroundCfg, UseTheme, Theme.
+ See /usr/share/doc/kdm/README.Debian for details
+ _
  Definition: the greeter is the login dialog, i.e., the part of &kdm;
  which the user sees.
  _
