Fixed upstream.Can be removed with next release.
--- a/kdm/kfrontend/CMakeLists.txt	2008/01/07 21:52:14	758435
+++ b/kdm/kfrontend/CMakeLists.txt	2008/01/07 22:01:53	758436
@@ -110,8 +110,6 @@
 build_confci()
 
 ## use 'GENKDMCONF_FLAGS=... make install' to override
-set(KDM_PATH "$ENV{DESTDIR}${CONFIG_INSTALL_DIR}/kdm")
-set(FACE_SRC_PATH "${CMAKE_CURRENT_SOURCE_DIR}/pics")
 install(CODE "
-exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --no-old --in '${KDM_PATH}' --no-in-notice --face-src '${FACE_SRC_PATH}' \$GENKDMCONF_FLAGS)
+exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --no-old --in \\\"\$DESTDIR${CONFIG_INSTALL_DIR}/kdm\\\" --no-in-notice --face-src \\\"${CMAKE_CURRENT_SOURCE_DIR}/pics\\\" \$GENKDMCONF_FLAGS)
 ")

