Index: b/kdm/config.def
===================================================================
--- a/kdm/config.def
+++ b/kdm/config.def
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 4
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/kde4/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
Index: b/kdm/kfrontend/kdmctl.c
===================================================================
--- a/kdm/kfrontend/kdmctl.c
+++ b/kdm/kfrontend/kdmctl.c
@@ -150,7 +150,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+	const char *cfg = "/etc/kde4/kdmrc";
 	char *ptr;
 	int fd;
 
Index: b/kdm/kfrontend/genkdmconf.c
===================================================================
--- a/kdm/kfrontend/genkdmconf.c
+++ b/kdm/kfrontend/genkdmconf.c
@@ -2766,6 +2766,7 @@
 
 
 static const char *oldkdes[] = {
+	"/etc/kde4",
 	KDE_CONFDIR,
 	"/opt/kde4/share/config",
 	"/usr/local/kde4/share/config",
@@ -2774,6 +2775,7 @@
 	"/usr/local/kde/share/config",
 	"/usr/local/share/config",
 	"/usr/share/config",
+	"/etc/kde3",
 
 	"/opt/kde3/share/config",
 	"/usr/local/kde3/share/config",
@@ -2817,7 +2819,7 @@
 "    Note that you possibly need to use --no-old-kde to make this take effect.\n"
 "  --old-kde /path/to/old/kde-config-dir\n"
 "    Where to look for the kdmrc of a previously installed kdm.\n"
-"    Default is to scan " KDE_CONFDIR " and\n"
+"    Default is to scan " "/etc/kde4" " and\n"
 "    {/usr,/usr/local,{/opt,/usr/local}/{kde4,kde,kde3}}/share/config.\n"
 "  --no-old\n"
 "    Do not look at older xdm/kdm configurations, just create default config.\n"
