We renamed ktrash to ktrash4. See bug #494427.
Index: kdebase-workspace/plasma/applets/trash/trash.cpp
===================================================================
--- kdebase-workspace.orig/plasma/applets/trash/trash.cpp	2008-09-06 11:24:09.000000000 +0200
+++ kdebase-workspace/plasma/applets/trash/trash.cpp	2008-09-06 11:24:40.000000000 +0200
@@ -193,7 +193,7 @@
         // we can simply call ktrash.
         //KonqOperations::emptyTrash(&m_menu);
         KProcess process;
-        process << KStandardDirs::findExe("ktrash") << "--empty";
+        process << KStandardDirs::findExe("ktrash4") << "--empty";
         process.execute();
 
     }
