Description: Provide a Xreset hook using a common framework.
Forwarded: not-needed
Bug-Debian: http://bugs.debian.org/571609
Author: Fathi Boudra <fabo@debian.org>

--- a/kdm/kfrontend/genkdmconf.c
+++ b/kdm/kfrontend/genkdmconf.c
@@ -625,6 +625,11 @@ static const char def_reset[] =
 "      exec sessreg -d -l \"$DISPLAY\" -u /var/run/utmp \\\n"
 "                   -h \"`echo $DISPLAY | cut -d: -f1`\" \"$USER\"\n"
 "      # NOTREACHED\n"
+"fi\n"
+"\n"
+"# Use common Xreset framework if it exist\n"
+"if [ -x /etc/X11/Xreset ] ; then\n"
+"   /etc/X11/Xreset\n"
 "fi\n";
 #endif /* _AIX */
 
