Index: kdebase-workspace-4.0.85+svn829067/kcminit/CMakeLists.txt
===================================================================
--- kdebase-workspace-4.0.85+svn829067.orig/kcminit/CMakeLists.txt	2008-07-07 20:31:49.000000000 +0000
+++ kdebase-workspace-4.0.85+svn829067/kcminit/CMakeLists.txt	2008-07-07 20:32:12.000000000 +0000
@@ -5,7 +5,7 @@
 
 kde4_add_kdeinit_executable( kcminit ${kcminit_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kcminit  ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kdeinit_kcminit  ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB})
 
 install(TARGETS kdeinit_kcminit ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kcminit         ${INSTALL_TARGETS_DEFAULT_ARGS} )
@@ -19,7 +19,7 @@
 
 kde4_add_kdeinit_executable( kcminit_startup ${kcminit_startup_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kcminit_startup  ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kdeinit_kcminit_startup  ${KDE4_KUTILS_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kcminit_startup ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kcminit_startup         ${INSTALL_TARGETS_DEFAULT_ARGS} )
Index: kdebase-workspace-4.0.85+svn829067/klipper/CMakeLists.txt
===================================================================
--- kdebase-workspace-4.0.85+svn829067.orig/klipper/CMakeLists.txt	2008-07-07 20:31:49.000000000 +0000
+++ kdebase-workspace-4.0.85+svn829067/klipper/CMakeLists.txt	2008-07-07 20:32:12.000000000 +0000
@@ -18,11 +18,16 @@
 
 kde4_add_kdeinit_executable(klipper ${klipper_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_klipper ${KDE4_KDEUI_LIBS})
+target_link_libraries(kdeinit_klipper ${KDE4_KDEUI_LIBS} ${ZLIB_LIBRARY}) 
 if (X11_Xfixes_FOUND)
   target_link_libraries(kdeinit_klipper ${X11_Xfixes_LIB})
 endif (X11_Xfixes_FOUND)
 
+if (X11_X11_LIB)
+  target_link_libraries(kdeinit_klipper ${X11_X11_LIB})
+endif (X11_X11_LIB)
+
+
 install(TARGETS kdeinit_klipper ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS klipper         ${INSTALL_TARGETS_DEFAULT_ARGS})
 
Index: kdebase-workspace-4.0.85+svn829067/ksmserver/CMakeLists.txt
===================================================================
--- kdebase-workspace-4.0.85+svn829067.orig/ksmserver/CMakeLists.txt	2008-07-07 20:31:49.000000000 +0000
+++ kdebase-workspace-4.0.85+svn829067/ksmserver/CMakeLists.txt	2008-07-07 20:32:12.000000000 +0000
@@ -47,7 +47,7 @@
 
 kde4_add_kdeinit_executable( ksmserver ${ksmserver_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_ksmserver plasma solidcontrol kworkspace ${KDE4_KDEUI_LIBS} ${QIMAGEBLITZ_LIBRARIES} ${X11_LIBRARIES})
+target_link_libraries(kdeinit_ksmserver plasma solidcontrol kworkspace ${KDE4_KDEUI_LIBS} ${QIMAGEBLITZ_LIBRARIES} ${X11_LIBRARIES} ${X11_Xrender_LIB})
 
 install(TARGETS kdeinit_ksmserver ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 
 
