--- a/kcminit/CMakeLists.txt
+++ b/kcminit/CMakeLists.txt
@@ -5,7 +5,7 @@
 
 kde4_add_kdeinit_executable( kcminit ${kcminit_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kcminit  ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kdeinit_kcminit  ${KDE4_KUTILS_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kcminit ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kcminit         ${INSTALL_TARGETS_DEFAULT_ARGS} )
@@ -19,7 +19,7 @@
 
 kde4_add_kdeinit_executable( kcminit_startup ${kcminit_startup_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kcminit_startup  ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kdeinit_kcminit_startup  ${KDE4_KUTILS_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kcminit_startup ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kcminit_startup         ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/klipper/CMakeLists.txt
+++ b/klipper/CMakeLists.txt
@@ -18,11 +18,15 @@
 
 kde4_add_kdeinit_executable(klipper ${klipper_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_klipper ${KDE4_KDEUI_LIBS})
+target_link_libraries(kdeinit_klipper ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} ${ZLIB_LIBRARY})
 if (X11_Xfixes_FOUND)
   target_link_libraries(kdeinit_klipper ${X11_Xfixes_LIB})
 endif (X11_Xfixes_FOUND)
 
+if (X11_X11_LIB)
+  target_link_libraries(kdeinit_klipper ${X11_X11_LIB})
+endif (X11_X11_LIB)
+
 install(TARGETS kdeinit_klipper ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS klipper         ${INSTALL_TARGETS_DEFAULT_ARGS})
 
--- a/kmenuedit/CMakeLists.txt
+++ b/kmenuedit/CMakeLists.txt
@@ -12,8 +12,7 @@
 
 kde4_add_kdeinit_executable( kmenuedit ${kmenuedit_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kmenuedit  ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTXML_LIBRARY}
-)
+target_link_libraries(kdeinit_kmenuedit  ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTXML_LIBRARY} ${QT_QTDBUS_LIBRARY} )
 
 install(TARGETS kdeinit_kmenuedit  DESTINATION ${LIB_INSTALL_DIR} )
 
--- a/ksysguard/gui/CMakeLists.txt
+++ b/ksysguard/gui/CMakeLists.txt
@@ -55,7 +55,7 @@
 
 kde4_add_kdeinit_executable(ksysguard ${ksysguard_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_ksysguard processui ${KDE4_KIO_LIBS} ksgrd plasma)
+target_link_libraries(kdeinit_ksysguard processui ${KDE4_KIO_LIBS} ksgrd plasma ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kdeinit_ksysguard ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS ksysguard         ${INSTALL_TARGETS_DEFAULT_ARGS})
--- a/kwin/kcmkwin/kwinoptions/CMakeLists.txt
+++ b/kwin/kcmkwin/kwinoptions/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(kcm_kwinoptions_PART_SRCS windows.cpp mouse.cpp main.cpp )
 kde4_add_plugin(kcm_kwinoptions ${kcm_kwinoptions_PART_SRCS})
-target_link_libraries(kcm_kwinoptions  ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QT3SUPPORT_LIBRARY})
+target_link_libraries(kcm_kwinoptions  ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QT3SUPPORT_LIBRARY} ${QT_QTDBUS_LIBRARY})
 install(TARGETS kcm_kwinoptions  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
 
--- a/kwin/lib/CMakeLists.txt
+++ b/kwin/lib/CMakeLists.txt
@@ -36,7 +36,7 @@
     )
 
 kde4_add_library(kwineffects SHARED ${kwin_EFFECTSLIB_SRCS})
-target_link_libraries(kwineffects ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${X11_LIBRARIES})
+target_link_libraries(kwineffects ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${X11_LIBRARIES} ${QT_QTDBUS_LIBRARY})
 set_target_properties(kwineffects PROPERTIES VERSION 1.0.0 SOVERSION 1 )
 
 install(TARGETS kwineffects ${INSTALL_TARGETS_DEFAULT_ARGS})
--- a/libs/kworkspace/CMakeLists.txt
+++ b/libs/kworkspace/CMakeLists.txt
@@ -17,7 +17,7 @@
 
 kde4_add_library(kworkspace SHARED ${kworkspace_LIB_SRCS})
 
-target_link_libraries(kworkspace  ${KDE4_KIO_LIBS} ${X11_LIBRARIES} )
+target_link_libraries(kworkspace  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_LIBRARIES} )
 
 set_target_properties(kworkspace PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS kworkspace ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/libs/plasma/CMakeLists.txt
+++ b/libs/plasma/CMakeLists.txt
@@ -80,7 +80,7 @@
 target_link_libraries(plasma ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} ${QT_QTUITOOLS_LIBRARY} ${QT_QTWEBKIT_LIBRARY})
 
 if(QT_QTOPENGL_FOUND AND OPENGL_FOUND)
-    target_link_libraries(plasma ${QT_QTOPENGL_LIBRARY} ${OPENGL_gl_LIBRARY})
+    target_link_libraries(plasma ${QT_QTOPENGL_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTSVG_LIBRARY} ${OPENGL_gl_LIBRARY} ${KDE4_SOLID_LIBS} )
 endif(QT_QTOPENGL_FOUND AND OPENGL_FOUND)
 
 set_target_properties(plasma PROPERTIES VERSION 1.0.0 SOVERSION 1)
--- a/libs/solid/control/CMakeLists.txt
+++ b/libs/solid/control/CMakeLists.txt
@@ -28,7 +28,7 @@
 
 kde4_add_library(solidcontrol SHARED ${solidcontrol_LIB_SRCS})
 
-target_link_libraries(solidcontrol ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} solidcontrolifaces )
+target_link_libraries(solidcontrol ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} solidcontrolifaces ${QT_QTDBUS_LIBRARY} ${QT_QTNETWORK_LIBRARY} )
 
 set_target_properties(solidcontrol PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS solidcontrol ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/libs/solid/control/backends/fakenet/CMakeLists.txt
+++ b/libs/solid/control/backends/fakenet/CMakeLists.txt
@@ -17,7 +17,7 @@
 kde4_add_plugin(solid_fakenet module.cpp ${solid_fakenet_PART_SRCS})
 
 
-target_link_libraries(solid_fakenet solidcontrolifaces ${KDE4_KDECORE_LIBS} ${QT_QTXML_LIBRARY} )
+target_link_libraries(solid_fakenet solidcontrolifaces ${KDE4_KDECORE_LIBS} ${QT_QTXML_LIBRARY} ${QT_QTNETWORK_LIBRARY} )
 
 install(TARGETS solid_fakenet DESTINATION ${PLUGIN_INSTALL_DIR}  )
 
--- a/libs/taskmanager/CMakeLists.txt
+++ b/libs/taskmanager/CMakeLists.txt
@@ -7,7 +7,7 @@
 
 kde4_add_library(taskmanager SHARED ${taskmanager_LIB_SRCS})
 
-target_link_libraries(taskmanager  ${KDE4_KDE3SUPPORT_LIBRARY} ${KDE4_KDEUI_LIBS} )
+target_link_libraries(taskmanager  ${KDE4_KDE3SUPPORT_LIBRARY} ${KDE4_KDEUI_LIBS} ${X11_X11_LIB} )
 if (X11_Xfixes_FOUND)
   target_link_libraries(taskmanager ${X11_Xfixes_LIB})
 endif (X11_Xfixes_FOUND)
--- a/solid/bluez/CMakeLists.txt
+++ b/solid/bluez/CMakeLists.txt
@@ -18,7 +18,7 @@
 
 kde4_add_plugin(solid_bluez module.cpp ${solid_bluez_PART_SRCS})
 
-target_link_libraries(solid_bluez ${KDE4_KDECORE_LIBS} ${QT_QTXML_LIBRARY} solidcontrolifaces)
+target_link_libraries(solid_bluez ${KDE4_KDECORE_LIBS} ${QT_QTXML_LIBRARY} ${QT_QTDBUS_LIBRARY} solidcontrolifaces)
 install(TARGETS solid_bluez DESTINATION ${PLUGIN_INSTALL_DIR})
 
 install(FILES solid_bluez.desktop DESTINATION ${SERVICES_INSTALL_DIR}/solidbackends )
--- a/solid/hal/CMakeLists.txt
+++ b/solid/hal/CMakeLists.txt
@@ -13,7 +13,7 @@
 
 kde4_add_plugin(solid_hal_power module_power.cpp ${solid_hal_power_PART_SRCS})
 
-target_link_libraries(solid_hal_power ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} solidcontrolifaces)
+target_link_libraries(solid_hal_power ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} ${QT_QTDBUS_LIBRARY} solidcontrolifaces)
 
 
 
--- a/kwin/kcmkwin/kwindecoration/CMakeLists.txt
+++ b/kwin/kcmkwin/kwindecoration/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(kcm_kwindecoration_PART_SRCS kwindecoration.cpp buttons.cpp preview.cpp )
 kde4_add_plugin(kcm_kwindecoration ${kcm_kwindecoration_PART_SRCS})
-target_link_libraries(kcm_kwindecoration ${KDE4_KDE3SUPPORT_LIBS} kdecorations ${X11_LIBRARIES})
+target_link_libraries(kcm_kwindecoration ${KDE4_KDE3SUPPORT_LIBS} kdecorations ${X11_LIBRARIES} ${QT_QTDBUS_LIBRARY})
 install(TARGETS kcm_kwindecoration  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
 ########### install files ###############
--- a/kwin/kcmkwin/kwinrules/CMakeLists.txt
+++ b/kwin/kcmkwin/kwinrules/CMakeLists.txt
@@ -10,7 +10,7 @@
 
 kde4_add_kdeinit_executable( kwin_rules_dialog ${kwin_rules_dialog_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kwin_rules_dialog ${KDE4_KDEUI_LIBS} )
+target_link_libraries(kdeinit_kwin_rules_dialog ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kwin_rules_dialog ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kwin_rules_dialog         ${INSTALL_TARGETS_DEFAULT_ARGS} )
@@ -22,7 +22,7 @@
 
 kde4_add_plugin(kcm_kwinrules ${kcm_kwinrules_PART_SRCS})
 
-target_link_libraries(kcm_kwinrules ${KDE4_KDEUI_LIBS} )
+target_link_libraries(kcm_kwinrules ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kcm_kwinrules  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kwin/clients/b2/CMakeLists.txt
+++ b/kwin/clients/b2/CMakeLists.txt
@@ -9,7 +9,7 @@
 
 kde4_add_plugin(kwin3_b2 ${kwin3_b2_PART_SRCS})
 
-target_link_libraries(kwin3_b2 kdecorations)
+target_link_libraries(kwin3_b2 kdecorations ${X11_X11_LIB})
 
 install(TARGETS kwin3_b2  DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/kwin/kcmkwin/kwincompositing/CMakeLists.txt
+++ b/kwin/kcmkwin/kwincompositing/CMakeLists.txt
@@ -11,7 +11,7 @@
     )
 kde4_add_ui_files(kcm_kwincompositing_PART_SRCS advanced.ui main.ui)
 kde4_add_plugin(kcm_kwincompositing ${kcm_kwincompositing_PART_SRCS})
-target_link_libraries(kcm_kwincompositing ${KDE4_KUTILS_LIBS})
+target_link_libraries(kcm_kwincompositing ${KDE4_KUTILS_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB})
 install(TARGETS kcm_kwincompositing  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
 # CompositingPrefs uses OpenGL
@@ -38,6 +38,9 @@
 if (X11_Xfixes_FOUND)
   target_link_libraries(kcm_kwincompositing ${X11_Xfixes_LIB})
 endif (X11_Xfixes_FOUND)
+if (X11_Xext_FOUND)
+  target_link_libraries(kcm_kwincompositing ${X11_Xext_LIB})
+endif (X11_Xext_FOUND)
 
 
 ########### install files ###############
--- a/kwin/kcmkwin/kwindesktop/CMakeLists.txt
+++ b/kwin/kcmkwin/kwindesktop/CMakeLists.txt
@@ -5,7 +5,7 @@
 
 kde4_add_plugin(kcm_kwindesktop ${kcm_kwindesktop_PART_SRCS})
 
-target_link_libraries(kcm_kwindesktop  ${KDE4_KDEUI_LIBS} ${KDE4_KUTILS_LIBS} )
+target_link_libraries(kcm_kwindesktop  ${KDE4_KDEUI_LIBS} ${KDE4_KUTILS_LIBS} ${X11_X11_LIB})
 
 install(TARGETS kcm_kwindesktop  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/plasma/applets/devicenotifier/CMakeLists.txt
+++ b/plasma/applets/devicenotifier/CMakeLists.txt
@@ -6,7 +6,7 @@
 
 kde4_add_ui_files(devicenotifier_SRCS deviceNotifierConfig.ui )
 kde4_add_plugin(plasma_applet_devicenotifier ${devicenotifier_SRCS})
-target_link_libraries(plasma_applet_devicenotifier plasma ${KDE4_KIO_LIBS})
+target_link_libraries(plasma_applet_devicenotifier plasma ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS plasma_applet_devicenotifier DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-applet-devicenotifier.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/plasma/applets/systemtray/CMakeLists.txt
+++ b/plasma/applets/systemtray/CMakeLists.txt
@@ -6,7 +6,7 @@
     systemtraywidget.cpp)
 
 kde4_add_plugin(plasma_applet_systemtray ${systemtray_SRCS})
-target_link_libraries(plasma_applet_systemtray ${KDE4_KDEUI_LIBS} plasma)
+target_link_libraries(plasma_applet_systemtray ${KDE4_KDEUI_LIBS} plasma ${X11_X11_LIB})
 
 install(TARGETS plasma_applet_systemtray DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-applet-systemtray.desktop DESTINATION ${SERVICES_INSTALL_DIR})
--- a/plasma/dataengines/dict/CMakeLists.txt
+++ b/plasma/dataengines/dict/CMakeLists.txt
@@ -5,7 +5,7 @@
 )
 
 kde4_add_plugin(plasma_engine_dict ${dict_engine_SRCS})
-target_link_libraries(plasma_engine_dict ${KDE4_KDECORE_LIBS} plasma)
+target_link_libraries(plasma_engine_dict ${KDE4_KDECORE_LIBS} plasma ${QT_QTNETWORK_LIBRARY})
 
 install(TARGETS plasma_engine_dict DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-dataengine-dict.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
--- a/kcontrol/keyboard/CMakeLists.txt
+++ b/kcontrol/keyboard/CMakeLists.txt
@@ -14,6 +14,7 @@
 
 
 target_link_libraries(kcm_keyboard
+${X11_X11_LIB}
     ${KDE4_KIO_LIBS} 
 )
 if(X11_XTest_FOUND)
--- a/kcontrol/kxkb/CMakeLists.txt
+++ b/kcontrol/kxkb/CMakeLists.txt
@@ -72,7 +72,7 @@
 
 kde4_add_plugin(kcm_keyboard_layout ${kcm_keyboard_layout_PART_SRCS})
 
-target_link_libraries(kcm_keyboard_layout ${KDE4_KIO_LIBS}
+target_link_libraries(kcm_keyboard_layout ${KDE4_KIO_LIBS} ${X11_X11_LIB}
     ${XKB_SUPPORT_LIB}
 )
 
@@ -97,7 +97,7 @@
 
 kde4_add_kdeinit_executable( kxkb ${kxkb_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kxkb ${X11_X11_LIB} ${X11_Xext_LIB} ${KDE4_KDEUI_LIBS}
+target_link_libraries(kdeinit_kxkb ${X11_X11_LIB} ${X11_Xext_LIB} ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY}
      ${XKB_SUPPORT_LIB})
 
 install(TARGETS kdeinit_kxkb ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/kcontrol/randr/CMakeLists.txt
+++ b/kcontrol/randr/CMakeLists.txt
@@ -33,7 +33,7 @@
 kde4_add_plugin(kcm_randr ${kcm_randr_PART_SRCS})
 
 
-target_link_libraries(kcm_randr  ${KDE4_KDEUI_LIBS} ${X11_Xrandr_LIB} )
+target_link_libraries(kcm_randr  ${KDE4_KDEUI_LIBS} ${X11_Xrandr_LIB} ${X11_X11_LIB} )
 
 install(TARGETS kcm_randr  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/plasma/dataengines/mouse/CMakeLists.txt
+++ b/plasma/dataengines/mouse/CMakeLists.txt
@@ -9,7 +9,7 @@
 endif (X11_Xfixes_FOUND)
 
 kde4_add_plugin(plasma_engine_mouse ${mouse_engine_SRCS})
-target_link_libraries(plasma_engine_mouse ${KDE4_KDEUI_LIBS} plasma)
+target_link_libraries(plasma_engine_mouse ${KDE4_KDEUI_LIBS} plasma ${X11_X11_LIB})
 if (X11_Xfixes_FOUND)
 	target_link_libraries(plasma_engine_mouse ${X11_Xfixes_LIB})
 endif (X11_Xfixes_FOUND)
--- a/plasma/dataengines/soliddevice/CMakeLists.txt
+++ b/plasma/dataengines/soliddevice/CMakeLists.txt
@@ -8,7 +8,7 @@
 )
 
 kde4_add_plugin(plasma_engine_soliddevice ${soliddevice_engine_SRCS})
-target_link_libraries(plasma_engine_soliddevice ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} plasma)
+target_link_libraries(plasma_engine_soliddevice ${KDE4_KDECORE_LIBS} ${KDE4_SOLID_LIBS} plasma ${QT_QTNETWORK_LIBRARY})
 
 install(TARGETS plasma_engine_soliddevice DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-dataengine-soliddevice.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
--- a/plasma/runners/bookmarks/CMakeLists.txt
+++ b/plasma/runners/bookmarks/CMakeLists.txt
@@ -6,7 +6,7 @@
 )
 
 kde4_add_plugin(krunner_bookmarksrunner ${krunner_bookmarksrunner_SRCS})
-target_link_libraries(krunner_bookmarksrunner ${KDE4_KDECORE_LIBS} ${QT_QTSCRIPT_LIBRARY} plasma)
+target_link_libraries(krunner_bookmarksrunner ${KDE4_KDECORE_LIBS} ${QT_QTSCRIPT_LIBRARY} plasma ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS krunner_bookmarksrunner DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/plasma/runners/webshortcuts/CMakeLists.txt
+++ b/plasma/runners/webshortcuts/CMakeLists.txt
@@ -3,7 +3,7 @@
 )
 
 kde4_add_plugin(krunner_webshortcuts ${krunner_webshortcuts_SRCS})
-target_link_libraries(krunner_webshortcuts ${KDE4_KIO_LIBS} plasma)
+target_link_libraries(krunner_webshortcuts ${KDE4_KIO_LIBS} plasma ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS krunner_webshortcuts DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/plasma/runners/xesam/CMakeLists.txt
+++ b/plasma/runners/xesam/CMakeLists.txt
@@ -3,7 +3,7 @@
 )
 
 kde4_add_plugin(krunner_xesam ${krunner_xesam_SRCS})
-target_link_libraries(krunner_xesam  plasma)
+target_link_libraries(krunner_xesam  plasma ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS krunner_xesam DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/access/CMakeLists.txt
+++ b/kcontrol/access/CMakeLists.txt
@@ -9,7 +9,7 @@
 kde4_add_plugin(kcm_access ${kcm_access_PART_SRCS})
 
 
-target_link_libraries(kcm_access  ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_KNOTIFYCONFIG_LIBS})
+target_link_libraries(kcm_access  ${KDE4_KIO_LIBS} ${KDE4_PHONON_LIBS} ${KDE4_KNOTIFYCONFIG_LIBS} ${X11_X11_LIB})
 
 install(TARGETS kcm_access  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
@@ -21,7 +21,7 @@
 
 kde4_add_kdeinit_executable( kaccess ${kaccess_KDEINIT_SRCS})
 
-target_link_libraries(kdeinit_kaccess  ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS})
+target_link_libraries(kdeinit_kaccess  ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS} ${X11_X11_LIB} )
 
 install(TARGETS kdeinit_kaccess ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(TARGETS kaccess         ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/kcontrol/colors/CMakeLists.txt
+++ b/kcontrol/colors/CMakeLists.txt
@@ -5,7 +5,7 @@
 kde4_add_ui_files(kcm_colors_PART_SRCS colorsettings.ui preview.ui)
 
 kde4_add_plugin(kcm_colors ${kcm_colors_PART_SRCS})
-target_link_libraries(kcm_colors ${KDE4_KFILE_LIBS} ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KDEUI_LIBS})
+target_link_libraries(kcm_colors ${KDE4_KFILE_LIBS} ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB})
 
 install(TARGETS kcm_colors  DESTINATION ${PLUGIN_INSTALL_DIR})
 install( FILES colors.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
--- a/kcontrol/energy/CMakeLists.txt
+++ b/kcontrol/energy/CMakeLists.txt
@@ -13,7 +13,7 @@
 kde4_add_plugin(kcm_energy ${kcm_energy_PART_SRCS})
 
 
-target_link_libraries(kcm_energy  ${KDE4_KIO_LIBS} )
+target_link_libraries(kcm_energy  ${KDE4_KIO_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} ${X11_Xext_LIB} )
 
 install(TARGETS kcm_energy  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/fonts/CMakeLists.txt
+++ b/kcontrol/fonts/CMakeLists.txt
@@ -11,7 +11,7 @@
 kde4_add_plugin(kcm_fonts ${kcm_fonts_PART_SRCS})
 
 
-target_link_libraries(kcm_fonts  ${KDE4_KDEUI_LIBS} ${FONTCONFIG_LIBRARIES} ${FREETYPE_LIBRARIES})
+target_link_libraries(kcm_fonts  ${KDE4_KDEUI_LIBS} ${FONTCONFIG_LIBRARIES} ${FREETYPE_LIBRARIES} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${X11_X11_LIB})
 
 install(TARGETS kcm_fonts  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/input/CMakeLists.txt
+++ b/kcontrol/input/CMakeLists.txt
@@ -42,6 +42,7 @@
 kde4_add_executable(kapplymousetheme ${kapplymousetheme_SRCS})
 
 target_link_libraries(kapplymousetheme 
+${X11_X11_LIB}
 ${KDE4_KDEUI_LIBS} ${X11_Xrender_LIB})
 if (X11_Xcursor_FOUND)
    target_link_libraries(kapplymousetheme ${X11_Xcursor_LIB})
@@ -60,7 +61,7 @@
 kde4_add_plugin(kcm_input ${kcm_input_PART_SRCS})
 
 
-target_link_libraries(kcm_input ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(kcm_input ${KDE4_KDE3SUPPORT_LIBS} ${X11_X11_LIB})
 if (LIBUSB_FOUND)
    target_link_libraries(kcm_input ${LIBUSB_LIBRARIES})
 endif (LIBUSB_FOUND)
--- a/kcontrol/kdm/CMakeLists.txt
+++ b/kcontrol/kdm/CMakeLists.txt
@@ -29,7 +29,7 @@
 kde4_add_plugin(kcm_kdm ${kcm_kdm_PART_SRCS})
 
 
-target_link_libraries(kcm_kdm ${KDE4_KDE3SUPPORT_LIBS} ${BLITZ_LIBRARIES} ${KDE4_KNEWSTUFF2_LIBS})
+target_link_libraries(kcm_kdm ${KDE4_KDE3SUPPORT_LIBS} ${BLITZ_LIBRARIES} ${KDE4_KNEWSTUFF2_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTSVG_LIBRARY} ${X11_X11_LIB})
 
 install(TARGETS kcm_kdm  DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/kcontrol/keys/CMakeLists.txt
+++ b/kcontrol/keys/CMakeLists.txt
@@ -13,7 +13,7 @@
 kde4_add_plugin(kcm_keys ${kcm_keys_PART_SRCS})
 
 
-target_link_libraries(kcm_keys  ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${X11_X11_LIB})
+target_link_libraries(kcm_keys  ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${X11_X11_LIB} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kcm_keys  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/launch/CMakeLists.txt
+++ b/kcontrol/launch/CMakeLists.txt
@@ -13,7 +13,7 @@
 kde4_add_plugin(kcm_launch ${kcm_launch_PART_SRCS})
 
 
-target_link_libraries(kcm_launch  ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY})
+target_link_libraries(kcm_launch  ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY})
 
 install(TARGETS kcm_launch  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
--- a/kcontrol/screensaver/CMakeLists.txt
+++ b/kcontrol/screensaver/CMakeLists.txt
@@ -26,7 +26,7 @@
 kde4_add_plugin(kcm_screensaver ${kcm_screensaver_PART_SRCS})
 
 
-target_link_libraries(kcm_screensaver  ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(kcm_screensaver  ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB})
 
 if(HAVE_GLXCHOOSEVISUAL)
    target_link_libraries(kcm_screensaver ${OPENGL_gl_LIBRARY})
--- a/kcontrol/style/CMakeLists.txt
+++ b/kcontrol/style/CMakeLists.txt
@@ -12,7 +12,7 @@
 
 kde4_add_plugin(kcm_style ${kcm_style_PART_SRCS})
 
-target_link_libraries(kcm_style ${KDE4_KDEUI_LIBS} ${BLITZ_LIBRARIES})
+target_link_libraries(kcm_style ${KDE4_KDEUI_LIBS} ${BLITZ_LIBRARIES} ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB})
 
 install(TARGETS kcm_style  DESTINATION ${PLUGIN_INSTALL_DIR})
 
--- a/khotkeys/libkhotkeysprivate/CMakeLists.txt
+++ b/khotkeys/libkhotkeysprivate/CMakeLists.txt
@@ -83,6 +83,9 @@
 )
 
 kde4_add_library(khotkeysprivate SHARED ${khotkeysprivate_SRCS})
+IF (X11_XTest_FOUND)
+    target_link_libraries(khotkeysprivate ${X11_XTest_LIB})
+ENDIF (X11_XTest_FOUND)
 target_link_libraries(khotkeysprivate kworkspace ${KDE4_KDE3SUPPORT_LIBS})
 set_target_properties(khotkeysprivate PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
 install(TARGETS khotkeysprivate DESTINATION ${LIB_INSTALL_DIR})
--- a/kcontrol/kfontinst/kcmfontinst/CMakeLists.txt
+++ b/kcontrol/kfontinst/kcmfontinst/CMakeLists.txt
@@ -9,7 +9,7 @@
 
 kde4_add_plugin(kcm_fontinst ${kcm_fontinst_PART_SRCS})
 
-target_link_libraries(kcm_fontinst ${KDE4_KIO_LIBS} ${KDE4_KDESU_LIBS} kfontinstui kfontinst )
+target_link_libraries(kcm_fontinst ${KDE4_KIO_LIBS} ${KDE4_KDESU_LIBS} kfontinstui kfontinst ${QT_QTDBUS_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kcm_fontinst  DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES fontinst.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
--- a/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt
+++ b/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt
@@ -2,6 +2,6 @@
 include_directories(${FREETYPE_INCLUDE_DIR})
 include_directories(${STRIGI_INCLUDE_DIR})
 kde4_add_library(font MODULE FontThroughAnalyzer.cpp FontEngine.cpp )
-target_link_libraries(font ${STREAMANALYZER_LIBRARY} kfontinst )
+target_link_libraries(font ${STREAMANALYZER_LIBRARY} ${STRIGI_STREAMANALYZER_LIBRARY} kfontinst )
 set_target_properties(font PROPERTIES  PREFIX strigita_)
 install(TARGETS font LIBRARY DESTINATION ${LIB_INSTALL_DIR}/strigi)
