From: Debian Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Subject: Comment out error message
Forwarded: not-needed
Last-Update: 2012-06-05
Index: kde-workspace/kdm/kfrontend/genkdmconf.c
===================================================================
--- kde-workspace.orig/kdm/kfrontend/genkdmconf.c	2013-09-27 09:16:47.891007661 +0000
+++ kde-workspace/kdm/kfrontend/genkdmconf.c	2013-09-27 09:16:47.887007817 +0000
@@ -3153,9 +3153,11 @@
                 memcmp(oldkde + olen - sizeof(SHR_CONF) + 1,
                        SHR_CONF, sizeof(SHR_CONF)))
             {
+                /*
                 fprintf(stderr,
                         "Warning: --old-kde does not end with " SHR_CONF ". "
                         "Might wreak havoc.\n");
+                */
                 oldkdepfx = oldkde;
             } else
                 ASPrintf((char **)&oldkdepfx,
