--- a/kdm/config.def
+++ b/kdm/config.def
@@ -52,11 +52,11 @@
 #endif
 
 #if defined(BSD) || defined(__linux__)
-# define DEF_USER_PATH KDM_DEF_USER_PATH
-# define DEF_SYSTEM_PATH KDM_DEF_SYSTEM_PATH
+# define DEF_USER_PATH "/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games"
+# define DEF_SYSTEM_PATH "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/bin/X11"
 #else
-# define DEF_USER_PATH KDM_DEF_USER_PATH ":/usr/ucb"
-# define DEF_SYSTEM_PATH KDM_DEF_SYSTEM_PATH ":/etc:/usr/ucb"
+# define DEF_USER_PATH "/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/usr/ucb"
+# define DEF_SYSTEM_PATH "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/bin/X11:/etc:/usr/ucb"
 #endif
 
 #if 0 /*def HASXDMAUTH*/
@@ -1372,13 +1372,13 @@
  This string is subject to word splitting.
  </para><para>
  The default is something reasonable for the system on which &kdm; was built,
- like <command>/usr/X11R6/bin/X</command>.
+ like <command>/usr/bin/X</command>.
 
 Key: ServerArgsLocal
 Type: string
 Default: ""
 User: core
-Instance: :*/"-nolisten tcp"
+Instance: :*/"-br -nolisten tcp"
 Comment: &
 Description:
  Additional arguments for the &X-Server;s for local sessions.
@@ -1932,10 +1932,10 @@
 
 Key: SessionsDirs
 Type: list
-Default: *KDMDATA "/sessions"
+Default: *"/usr/share/xsessions,/var/lib/menu-xdg/xsessions,/usr/share/apps/kdm/sessions"
 User: core
 User: greeter-c
-Instance: */KDMCONF "/sessions," KDMDATA "/sessions"
+Instance: #*/"/usr/share/xsessions,/var/lib/menu-xdg/xsessions,/usr/share/apps/kdm/sessions"
 Update: upd_sessionsdirs
 Comment:
  The directories containing session type definitions in .desktop format,
@@ -2004,7 +2004,7 @@
 
 Key: UseSessReg
 Type: bool
-Default: true
+Default: false
 User: core
 Instance: #*/!
 Comment:
@@ -2095,9 +2095,9 @@
 
 Key: GreetString
 Type: string
-Default: "Welcome to %s at %n"
+Default: "Welcome to Debian at %n"
 User: greeter
-Instance: #*/"K Desktop Environment (%n)"
+Instance: #*/"Welcome to Debian at %n"
 Comment:
  The headline in the greeter. The following character pairs are replaced:
  - %d -> current display
@@ -2556,7 +2556,6 @@
 Default: 0
 User: greeter
 Instance: #*/
-Update: upd_forgingseed
 Comment:
  Random seed for forging saved session types, etc. of unknown users.
  This value should be random but constant across the login domain.
@@ -2653,7 +2652,7 @@
 Type: bool
 Default: true
 User: greeter
-Instance: :*/false
+Instance: :*/true
 Comment: &
 Description:
  Show the <guilabel>Restart X Server</guilabel>/<guilabel>Close Connection</guilabel> action in the greeter.
