Description: KRunner patch so it detects Iceweasel bookmarks
Author: Guido Loupias <guidol@lavabit.com>
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=646612
Forwarded: not-needed
Reviewed-by: Maximiliano Curia <maxy@debian.org>
Last-Update: 2013-07-15
Index: kde-workspace/plasma/generic/runners/bookmarks/browserfactory.cpp
===================================================================
--- kde-workspace.orig/plasma/generic/runners/bookmarks/browserfactory.cpp	2013-07-15 14:08:42.713972459 +0200
+++ kde-workspace/plasma/generic/runners/bookmarks/browserfactory.cpp	2013-07-15 14:09:11.008690868 +0200
@@ -33,7 +33,7 @@
     }
     delete m_previousBrowser;
     m_previousBrowserName = browserName;
-    if (browserName.contains("firefox", Qt::CaseInsensitive)) {
+    if (browserName.contains("firefox", Qt::CaseInsensitive) || browserName.contains("iceweasel", Qt::CaseInsensitive)) {
         m_previousBrowser = new Firefox(parent);
     } else if (browserName.contains("opera", Qt::CaseInsensitive)) {
         m_previousBrowser = new Opera(parent);
